### PR TITLE
Cli and config

### PR DESCRIPTION
**Pull request by skermes**

### CODE REVIEW

**Comment by skermes** [comment]
Don't review this until I get a chance to fix the bug in it.

**Comment by skermes** [comment]
Bugs I knew about are fixed now, feel free to review.

File: src/main/java/com/ripariandata/timberwolf/conf4j/ConfigFileParser.java [hunk 1]
```diff
@@ -51,7 +53,16 @@ public ConfigFileParser(final Object bean)
```
Comment (ScottDugas) [inline src/main/java/com/ripariandata/timberwolf/conf4j/ConfigFileParser.java:53]: What's a bean?

File: src/main/java/com/ripariandata/timberwolf/conf4j/NonOverwritingFieldSetter.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+package com.ripariandata.timberwolf.conf4j;
+
+import java.lang.reflect.Field;
+
+/**
+ * NonOverwritingFieldSetter wraps around a single field in a single object, so
+ * so that later we can set the value of that field.  Calling <tt>set</tt> on a
+ * field whose value is not the default value for the field's type will do nothing.
+ * <p>
+ * The default value is 0 for numeric types, false for booleans, '\u0000' for chars,
```
Comment (ScottDugas) [inline src/main/java/com/ripariandata/timberwolf/conf4j/NonOverwritingFieldSetter.java]: the default is the value set on the field when this object is created. Declared variables that are not assigned to, have the values: ...

File: src/main/java/com/ripariandata/timberwolf/conf4j/NonOverwritingFieldSetter.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                throw new IllegalAccessError(iae.getMessage());
+            }
+        }
+        return val;
+    }
+
+    /**
+     * Sets the field represented by this setter to the given value as long as
+     * the field currently contains the default value for its type.
```
Comment (ScottDugas) [inline src/main/java/com/ripariandata/timberwolf/conf4j/NonOverwritingFieldSetter.java]: Again, it's the value when this FieldSetter was created, not the default for the type.

**Comment by ScottDugas** [comment]
What happens if I have the following in my config file:
`hbase.key.header="sender"`
but I want it to be "Item ID" (currently the default for that config entry
if I pass the argument:
`--hbase-column-family "Item ID"`
the Arg parser will set the field to "Item ID", and when config file parser goes to set it, it will confirm that it's the default and then set it to "sender"

**Comment by skermes** [comment]
Ultimately, the things I was trying to do here worked, but made things more complicated than we really wanted them to be.  Merging the arguments around like that was adding more complexity than it was removing.

I've tagged the last commit on this branch as `complex-argument-handling`, if we ever decide to come back to it.

**Pull request closed without merging**